rohmu/object_storage/swift.py

```diff
@@ -297,17 +297,24 @@ def copy_file(self, *, source_key, destination_key, metadata=None, **_kwargs):
 
     def store_file_object(self, key, fd, *, cache_control=None, metadata=None, mimetype=None, upload_progress_fn=None):
         metadata = metadata or {}
+        notify_size = [0]
+
+        def upload_progress_fn_wrapper(bytes_written):
+            notify_size[0] = bytes_written
+            if upload_progress_fn:
+                upload_progress_fn(bytes_written)
+
         self._store_file_contents(
             key,
             fd,
             cache_control=cache_control,
             metadata=metadata,
             mimetype=mimetype,
-            upload_progress_fn=upload_progress_fn,
+            upload_progress_fn=upload_progress_fn_wrapper,
             multipart=True,
             content_length=metadata.get("Content-Length"),
         )
-        self.notifier.object_created(key=key, size=os.path.getsize(fd))
+        self.notifier.object_created(key=key, size=notify_size[0])
```

Comment:
Since the size will only be available when metadata.get("Content-Length") is available, why not using that directly instead of wrapping the callback just to get back the same value ?